tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c

```diff
@@ -840,44 +919,202 @@ static void test_scanx_main(void)
 	}
 
 	printk("Start scanning...");
+	is_periodic = false;
+	per_adv_evt_cnt_actual = 0;
+	per_adv_evt_cnt_expected = 3;
 	err = bt_le_scan_start(&scan_param, scan_cb);
 	if (err) {
 		goto exit;
 	}
 	printk("success.\n");
 
-	/* TODO: Replace sleep with verification of adv reports and number
-	 * of reports expected.
-	 */
-	k_sleep(K_SECONDS(10));
+	printk("Waiting...");
+	while (!is_periodic ||
+	       (per_adv_evt_cnt_actual != per_adv_evt_cnt_expected)) {
+		k_sleep(K_MSEC(30));
+	}
+	printk("done.\n");
 
-#if TEST_LOW_LEVEL
-	uint8_t type = (BIT(0) << 1) | 0x01; /* 1M PHY and active scanning */
+	printk("Stop scanning...");
+	err = bt_le_scan_stop();
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
 
-	printk("Setting scan parameters...");
-	err = ll_scan_params_set(type, SCAN_INTERVAL, SCAN_WINDOW,
-				 OWN_ADDR_TYPE, FILTER_POLICY);
+	printk("Creating Periodic Advertising Sync 0...");
+	bt_addr_le_copy(&sync_create_param.addr, &per_addr);
+	sync_create_param.options = 0;
+	sync_create_param.sid = 0xf;
+	sync_create_param.skip = 0;
+	sync_create_param.timeout = 0xa;
+	err = bt_le_per_adv_sync_create(&sync_create_param, &sync_cb, &sync);
```

Comment:
Make sure to update this to the new API (https://github.com/zephyrproject-rtos/zephyr/pull/28730) when it's merged

Reply:
Sure. Lets get yours merged first, I will rebase and test.

Reply:
@cvinayak The PR has been merged